src/kernel/crane_constructor/hook_chooser/hook.rs

```diff
@@ -0,0 +1,179 @@
+use crate::kernel::storage::storage::Storage;
+use crate::kernel::storage::storage::Value;
+use crate::Param_to_compare;
+use std::io;
+pub struct Hook{
+    // Vec<Vec<String> Крюки<Характеристики>
+
+    pub hook: Vec<String>,
+    pub bearing: String
+}
+
+
+
+impl Hook{
+
+    pub fn new(_param_comp: Param_to_compare, hooks_storage: &mut Storage) -> Self {
+
+        println!("{}",_param_comp._hook_type);
```

Comment:
Use `log::info / debug / warn / error` instead
add `dbgid` to your classes
```rust
struct Hook {
    dbgid: String,
}
```

Then use it in the debugging like
Also describe each debugged value with it exact name
```rust
log::debug!("{}.new | hook_type: {}",_param_comp._hook_type);
```